testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js

```diff
@@ -1678,6 +1680,84 @@ QUnit.module('Options', {
+                this.cacheTableElements();
+
+                assert.strictEqual(this.$headerRow.children().length, expectedColumnCount);
+                assert.strictEqual(this.$firstBodyRow.children().length, expectedColumnCount);
+            };
+        },
+    }, () => {
+        QUnit.test('table should have additional column if showWeekNumbers=true', function() {
+            this.reinit({ showWeekNumbers: true });
+
+            this.checkColumnCount(8);
+        });
+
+        QUnit.test('table should not have additional column if showWeekNumbers=false', function() {
+            this.reinit({ showWeekNumbers: false });
+
+            this.checkColumnCount(7);
+        });
+
+        QUnit.test('table should be rerendred with additional column after runtime chage of showWeekNumbers', function(assert) {
+            this.reinit({});
+
+            this.checkColumnCount(7);
+
+            this.calendar.option('showWeekNumbers', true);
+
+            this.checkColumnCount(8);
+        });
+
+        QUnit.test('first header cell should have "dx-week-number-header" class when showWeekNumbers=true', function(assert) {
+            this.reinit({ showWeekNumbers: true });
+            this.cacheTableElements();
+            const $firstHeaderCell = this.$headerRow.children().eq(0);
+
+            assert.ok($firstHeaderCell.hasClass(CALENDAR_WEEK_NUMBER_HEADER_CLASS));
+        });
+
+        QUnit.test('first cell in tbody should have "dx-calendar-week-number-cell" class when showWeekNumbers=true', function(assert) {
+            this.reinit({ showWeekNumbers: true });
+            this.cacheTableElements();
+            const $firstBodyCell = this.$firstBodyRow.children().eq(0);
+
+            assert.ok($firstBodyCell.hasClass(CALENDAR_WEEK_NUMBER_CELL_CLASS));
+        });
+
+        QUnit.test('last header cell should have "dx-week-number-header" class when showWeekNumbers=true and rtlEnabled=true', function(assert) {
+            this.reinit({ showWeekNumbers: true, rtlEnabled: true });
+            this.cacheTableElements();
+            const $lastHeaderCell = this.$headerRow.children().eq(7);
+
+            assert.ok($lastHeaderCell.hasClass(CALENDAR_WEEK_NUMBER_HEADER_CLASS));
+        });
+
+        QUnit.test('last cell in tbody should have "dx-calendar-week-number-cell" class when showWeekNumbers=true and rtlEnabled=true', function(assert) {
+            this.reinit({ showWeekNumbers: true, rtlEnabled: true });
+            this.cacheTableElements();
+            const $lastBodyRowCell = this.$firstBodyRow.children().eq(7);
+
+            assert.ok($lastBodyRowCell.hasClass(CALENDAR_WEEK_NUMBER_CELL_CLASS));
+        });
+
+        QUnit.test('calendar with zoomLevel!=="month" and showWeekNumbers=true should not have additional column', function(assert) {
+            this.reinit({ showWeekNumbers: true, zoomLevel: 'year' });
+            this.cacheTableElements();
+
+            assert.ok(true);
+        });
+    });
```

Comment:
What about accessibility? Did u test it?

We should add tests.

Reply:
added test on aria-label and role of week number cells